src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -56,7 +56,7 @@ public class LockFileBuilder
             }
 
             // Record all libraries used
-            foreach (var item in targetGraphs.SelectMany(g => g.Flattened).Distinct()
+            foreach (var item in targetGraphs.SelectMany(g => g.Flattened).Distinct(GraphItemKeyComparer<RemoteResolveResult>.Instance)
```

Comment:
Are there non-CPVM scenarios where this will change behaviour?

If I understand correctly the fix is to change this comparison from checking both key and data, to using only the key?

Reply:
No impact on non-CPVM scenarios. Data was never included in the [equality ](https://github.com/NuGet/NuGet.Client/blob/bb4f636278eeebb2791eb97a65324ca74760b1e7/src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphItem.cs#L32).